### DISCUSSION TITLE
Limit CPU frequency

### DISCUSSION THREAD

**Discussion by rrussell263** [seerge/g-helper#1907] [Ideas]
Is it at all possible to limit CPU frequency to a set number, as my laptop G713PI is stable at 3GHZ but once i enable the boost, it dies around 5.4ghz with a black screen reset. I have tried limiting the CPU to 40w, 20w, and 60w, but each time one of the cores hits that threshold and it forces a reboot. However, when all cores are running at once, and the CPU speed is limited in the windows power options to 75% max, its fine and stable until idle, when it does the reset- leading me to theorize that the CPU isn't able to handle 5.4, but able to handle something closer to 4.8 single core.

**Comment by seerge** [comment]
@rrussell263 hello, why don't you just stick to boost off ?

You can also try to limit CPU clock via windows power plan (same place where boost option is), but from my experience it simply has no effect on modern devices at all.

You can't limit clock anyhow else

**Reply by IceStormNG** [reply]
Boost off heavily limits Intel CPUs. Throttlestop can do it by setting the turbo ratio MSR.
The power plan settings does work, but only if the register is unlocked in the firmware (CFG Lock and OC Lock).
I know how to do that even for machines that do not have the setting visible in the BIOS and the power plan setting does work then.

Maybe, we could also add that setting to GHelper. Would be just two sliders (one for battery and one when on charger or we just do one slider for both).

@rrussell263  can you please try to use the tool PowerSettingsExplorer (https://forums.guru3d.com/threads/windows-power-plan-settings-explorer-utility.416058/) and set the max frequency for both efficiency classes to the desired value, and then observe whether the CPU respects that setting or not. The value is in Mhz.

On my 13900H M16 it does work. However, I do have the firmware unlocked, so I'm not sure whether this works without this.

**Reply by ev-i** [reply]
On my Asus TUF Dash F15 FX516PR setting Maximum Processor Frequency works.
![image](https://github.com/seerge/g-helper/assets/47784335/ff7a4d33-6517-4e3f-9b2c-5c61656d848c)
It would be very useful for me if I could set different max frequency for each mode (silent, balanced, turbo)

**Reply by seerge** [reply]
@ev-i that's great :) You can just create a separate power plan with limited clocks (by copying current) and assign it to any mode in G-Helper in config as explained here https://github.com/seerge/g-helper/wiki/Power-user-settings#custom-windows-power-plans-with-each-mode.

I don't want to add this to the app. As this will only raise a new wave of hate, as people will see feature and accuse G-Helper in "limiting their clocks". Good example is, since G-Helper started showing battery wear (by just reading it from windows), people accuse it in ... "destroying and messing their battery".

Over last time I have read so much shit, lies and accusation in my address, so I don't want to add more performance related features.

**Reply by rrussell263** [reply]
Tried to use the frequency limiter with that tool to no avail. I have a 7845HX.

**Reply by s4-Liuzhuang** [reply]
<img width="706" alt="image" src="https://github.com/seerge/g-helper/assets/143558311/e768f253-6cbc-4e2e-8d1f-d0d2ebb31fdc">
why does it not work for me

**Reply by maa7** [reply]
> I don't want to add this to the app. As this will only raise a new wave of hate, as people will see feature and accuse G-Helper in "limiting their clocks".

I understand that, but still, if it's possible to create a processor frequency limit slider, please do it. (It might even be possible to display this slider through a secret option in the settings.)

**Reply by maa7** [reply]
> 
> You can also try to limit CPU clock via windows power plan (same place where boost option is), but from my experience it simply has no effect on modern devices at all.
> 

I have a modern laptop with a Ryzen 9 9955HX processor, and the frequency limitation works perfectly.
However, you need to edit the config.json file to assign each Windows Power Plan to each G‑Helper power mode so that it’s possible to limit the processor frequency in each mode.

**Comment by Kikimahe** [comment]
> can you please try to use the tool PowerSettingsExplorer (https://forums.guru3d.com/threads/windows-power-plan-settings-explorer-utility.416058/) and set the max frequency for both efficiency classes to the desired value, and then observe whether the CPU respects that setting or not. The value is in Mhz.

@IceStormNG 

Hi 

I tested my Intel 12700H and PowerSettingsExplorer to display "max CPU frequency" in power options.

It works. :)

But
1. the defined values (windows) are not respected precisely (Hwinfo). 
2. there are 2 different settings to modify :  1 for P-core and 1 for E-Core.



In Windows power options.
`“Max frequency Cpu”` is for **E-Core.**
- Value 1200 Mhz in Windows, displays 898 Mhz in Hwinfo
- Value 600 Mhz in windows, displays 399 Mhz in Hwinfo

`“Max frequency CPU corresponding to energy efficiency class 1”` is for **P-Core**
- Value 3200 Mhz in Windows, displays 3092 Mhz in Hwinfo
- Value 2000 Mhz in windows, displays 1895 Mhz in Hwinfo

So I think that there are pre-defined levels (Mhz) somewhere because this does not precisely respect the values indicated in Max frequency P-Core and E-core.
 


Example of CPU limitation and stress test :
![Capture d'écran 2024-01-24 112854](https://github.com/seerge/g-helper/assets/142881210/e1ca1e07-af64-42a9-a23b-a120cf232d19)

**Reply by IceStormNG** [reply]
Yeah, I also noticed that the CPU usually stays below the set limit. This could be due to slight inaccuracies in the base clock (not being 100Mhz but 99.5Mhz or so). And the CPU wants to stay below that.
I don't see this as a problem. 100-200Mhz difference is not gonna do anything noticeable. But limiting my 13900H to 4.4Ghz (down from 5.whatever) is for sure noticeable in heat and power draw. Performance.... is more or less the same. I really cannot tell a difference.

**Reply by Anres1981** [reply]
`
powercfg /SETDCVALUEINDEX SCHEME_CURRENT SUB_PROCESSOR PROCFREQMAX 3500

powercfg /SETACVALUEINDEX SCHEME_CURRENT SUB_PROCESSOR PROCFREQMAX 3500
`

**Reply by syedtufailasuspro** [reply]
> > can you please try to use the tool PowerSettingsExplorer (https://forums.guru3d.com/threads/windows-power-plan-settings-explorer-utility.416058/) and set the max frequency for both efficiency classes to the desired value, and then observe whether the CPU respects that setting or not. The value is in Mhz.
> 
> @IceStormNG
> 
> Hi
> 
> I tested my Intel 12700H and PowerSettingsExplorer to display "max CPU frequency" in power options.
> 
> It works. :)
> 
> But
> 
>     1. the defined values (windows) are not respected precisely (Hwinfo).
> 
>     2. there are 2 different settings to modify :  1 for P-core and 1 for E-Core.
> 
> 
> In Windows power options. `“Max frequency Cpu”` is for **E-Core.**
> 
>     * Value 1200 Mhz in Windows, displays 898 Mhz in Hwinfo
> 
>     * Value 600 Mhz in windows, displays 399 Mhz in Hwinfo
> 
> 
> `“Max frequency CPU corresponding to energy efficiency class 1”` is for **P-Core**
> 
>     * Value 3200 Mhz in Windows, displays 3092 Mhz in Hwinfo
> 
>     * Value 2000 Mhz in windows, displays 1895 Mhz in Hwinfo
> 
> 
> So I think that there are pre-defined levels (Mhz) somewhere because this does not precisely respect the values indicated in Max frequency P-Core and E-core.
> 
> Example of CPU limitation and stress test : ![Capture d'écran 2024-01-24 112854](https://private-user-images.githubusercontent.com/142881210/299283960-e1ca1e07-af64-42a9-a23b-a120cf232d19.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0MzA0MDUsIm5iZiI6MTcxOTQzMDEwNSwicGF0aCI6Ii8xNDI4ODEyMTAvMjk5MjgzOTYwLWUxY2ExZTA3LWFmNjQtNDJhOS1hMjNiLWExMjBjZjIzMmQxOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyNlQxOTI4MjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MWZhMGU2YTFkODUwNGY0NGYxMDYwNWI2ZjJiMDI3MzBlNTYyMjhkMjQ3OWEzODUzZWZkMDRkNzhlYmFjZTRkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.lwam3vHXp_ZcbtGFQbLB8x8J1XY-1AKHNZ0cPHPUEAE)

@Kikimahe I have an intel 12450H . I tried exactly what you told but it does not work. I don't know why. May be u could help me in this regard.
